src/components/Form.js

```diff
@@ -1,51 +1,85 @@
-import React, {useState} from "react";
-function Form({addDetails,person, setPerson}){
+import React from "react";
+const Form = ({propsObject}) => {
+    const {detailsList,setPersonDetails, personDetails,addDetails,isEditButtonClicked,setEditButton,setDetailsList, personId} = propsObject 
     const handleClick = (e) => {
         e.preventDefault();
-        addDetails(person.name, person.fatherName, person.age, person.address);
-        setPerson({name: "", fatherName: "", age: 0, address: ""})
+        
+        if(personDetails.name==="" || personDetails.fatherName === "" || personDetails.address ==="" || personDetails.age===0) {
```

Comment:
Clean your conditional statement e.g `!personDetails.name` instead of `personDetails.name===""`